src/NuGet.Core/NuGet.Commands/SignCommand/CertificateFindOptions.cs

```diff
@@ -42,10 +42,15 @@ internal class CertificateSourceOptions
         public string SubjectName { get; set; }
 
         /// <summary>
-        /// The SHA-1 fingerprint of the certificate.
+        /// SHA256 or SHA384 or SHA512 fingerprint of the certificate.
         /// </summary>
         public string Fingerprint { get; set; }
 
+        /// <summary>
+        /// The SHA256 or SHA384 or SHA512 hash algorithm name used to calculate the fingerprint of the certificate.
+        /// </summary>
+        public HashAlgorithmName FingerprintHashAlgorithmName { get; set; }
```

Comment:
We are deducing the algorithm from the fingerprint length. https://github.com/NuGet/NuGet.Client/pull/5895/files#diff-d1848c7de42ac59aab4c7b5b1e4fcacf835ad5aed1018d6bdff8445d0d34433dR392-R422

The reason why I added this new property is because, if it is SHA-1 the following code works 
```cs 
store.Certificates.Find(X509FindType.FindByThumbprint, options.Fingerprint, validOnly);
```
Whereas for SHA-256, 384 and 512 we have to loop through the certificates to locate the certificate from the store.

https://github.com/NuGet/NuGet.Client/pull/5895/files#diff-50e182f9a08392f15c7d8229710796712bcaf61981d35eac78fbb2aa9665703cR180-R188


```cs
                    foreach (var cert in store.Certificates)
                    {
                        string actualFingerprint = CertificateUtility.GetHashString(cert, options.FingerprintHashAlgorithmName);

                        if (string.Equals(actualFingerprint, options.Fingerprint, StringComparison.InvariantCultureIgnoreCase))
                        {
                            resultCollection.Add(cert);
                            break;
                        }
                    }
```

Reply:
I thought it would be helpful to explain why I chose to deduce the hash algorithm based on the certificate fingerprint while parsing arguments passed to the command line.

1. If an invalid certificate fingerprint is passed, the command will exit early instead of invoking the core library. This means we have to deduce the hash algorithm while parsing the arguments.
2. Instead of deducing the hash algorithm while searching for the certificate fingerprint, I chose to add this property so that it can be used in the core library API.

Reply:
Removed this new property in the latest commit.